src/main/java/org/datadog/jmxfetch/RemoteConnection.java

```diff
@@ -89,6 +93,18 @@ public RemoteConnection(Map<String, Object> connectionParams) throws IOException
         // Set an RMI timeout so we don't get stuck waiting for a bean to report a value
         System.setProperty("sun.rmi.transport.tcp.responseTimeout", rmiTimeout);
 
+        Integer rmiConnectionTimeout = DEFAULT_RMI_CONNECTION_TIMEOUT;
+        try {
+            rmiConnectionTimeout = (Integer) connectionParams.get("rmi_connection_timeout");
+        } catch (final ClassCastException e) {
+            rmiConnectionTimeout = 
+                Integer.parseInt((String) connectionParams.get("rmi_connection_timeout"));
```

Comment:
note to self: this would not trigger a `java.lang.NumberFormatException` when `connectionParams.get("rmi_connection_timeout")` is null, because the cast of `(Integer) null` is valid. So we're good.